File: app/src/main/java/com/example/roombooking/utils/CommonUtils.java
package com.example.roombooking.utils;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.view.View;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class CommonUtils
{
	private static final String ROOM_PREFIX = "FH";
	private static final String EMAIL_POSTFIX = "fh-bielefeld.de";

	/**
	 * do validate email address
	 *
	 * @param emailAddress : the email address need to validate
	 * @return : true if that email is valid, false if otherwise
	 */
	public static boolean isEMailValid(String emailAddress)
	{
		if (emailAddress != null && !emailAddress.isEmpty())
		{
			String regex = "^[\\w!#$%&'*+/=?`{|}~^-]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$";
			Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
			Matcher matcher = pattern.matcher(emailAddress);

			if(matcher.matches())
			{
				String[] emailParts = emailAddress.split("@");
				return emailParts[1].equals(EMAIL_POSTFIX);
			}
			return false;
		}
		return false;
	}

	public static boolean isStartTimeValid(long timeMillis)
	{
		return timeMillis > System.currentTimeMillis();
	}

	/**
	 * do validate email address
	 *
	 * @param qrString : the String generated from QRCode need to validate
	 */
	public static boolean isQRCodeValid(String qrString)
	{
		return qrString != null && qrString.length() > 2 && (qrString.substring(0, 2).equals(ROOM_PREFIX));
	}

	/**
	 * do check internet connection's availability
	 *
	 * @param context
	 * @return : true if connected, false if otherwise
	 */
	public static boolean isNetworkAvailable(Context context)
	{
		ConnectivityManager connectivityManager
				= (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
		NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();

		if(activeNetworkInfo != null)
			return activeNetworkInfo.isConnected();

		return false;
	}

	/**
	 * do hide soft keyboard
	 *
	 * @param activity: current Activity in which soft keyboard should be hidden
	 */
	public static void hideSoftKeyboard (Activity activity)
	{
		if (activity != null)
		{
			InputMethodManager inputMethodManager = (InputMethodManager) activity
					.getSystemService(Context.INPUT_METHOD_SERVICE);
			View viewFocus = activity.getCurrentFocus();
			if (viewFocus != null)
			{
				inputMethodManager.hideSoftInputFromWindow(viewFocus.getWindowToken(), 0);
			}
			activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
		}
	}

}

File: app/src/main/java/com/example/roombooking/utils/CommonUtils.kt
package com.example.roombooking.utils

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager
import android.view.WindowManager
import android.view.inputmethod.InputMethodManager
import java.text.SimpleDateFormat
import java.util.*
import java.util.regex.Pattern

object CommonUtils {
    private const val ROOM_PREFIX = "FH"
    private const val EMAIL_POSTFIX = "fh-bielefeld.de"

    /**
     * do validate email address
     *
     * @param emailAddress : the email address need to validate
     * @return : true if that email is valid, false if otherwise
     */
	 @JvmStatic
	 fun isEMailValid(emailAddress: String): Boolean {
        if (emailAddress.isNotEmpty()) {
            val regex = "^[\\w!#$%&'*+/=?`{|}~^-]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$"
            val pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE)
            val matcher = pattern.matcher(emailAddress)
            if (matcher.matches()) {
                val emailParts = emailAddress.split("@").toTypedArray()
                return emailParts[1] == EMAIL_POSTFIX
            }
            return false
        }
        return false
    }

//    fun isStartTimeValid(timeMillis: Long): Boolean {
//        return timeMillis > System.currentTimeMillis()
//    }

    /**
     * do validate email address
     *
     * @param qrString : the String generated from QRCode need to validate
     */
	 @JvmStatic
	 fun isQRCodeValid(qrString: String?): Boolean {
        return qrString != null && qrString.length > 2 && qrString.substring(0, 2) == ROOM_PREFIX
    }

    /**
     * do check internet connection's availability
     *
     * @param context
     * @return : true if connected, false if otherwise
     */
	 @JvmStatic
	 fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = connectivityManager.activeNetworkInfo
        return activeNetworkInfo?.isConnected ?: false
    }

    /**
     * do hide soft keyboard
     *
     * @param activity: current Activity in which soft keyboard should be hidden
     */
	 @JvmStatic
	 fun hideSoftKeyboard(activity: Activity?) {
        if (activity != null) {
            val inputMethodManager = activity
                    .getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            val viewFocus = activity.currentFocus
            if (viewFocus != null) {
                inputMethodManager.hideSoftInputFromWindow(viewFocus.windowToken, 0)
            }
            activity.window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)
        }
    }

    @JvmStatic
	 val currentDateTime: String
        get() = SimpleDateFormat("dd/MM/yyyy HH:mm").format(Calendar.getInstance().time)
}